### ISSUE TITLE
Could not download zip file for zurb foundation

### ISSUE THREAD

**Issue by pbuyle** [drupal-composer/drupal-packagist#37]
When using `composer install` on a project including [ZURB Foundation](https://www.drupal.org/project/zurb_foundation), composer tries to donwload http://ftp.drupal.org/files/projects/zurb-foundation-7.x-5.0-rc6.zip (with a `-` between zurb and foundation), but the proper URL is http://ftp.drupal.org/files/projects/zurb_foundation-7.x-5.0-rc6.zip (with an `_` between zurb and foundation).

```
Installing drupal/zurb_foundation (7.5.0-rc6)
   Downloading: 0%    Failed to download drupal/zurb_foundation from dist: The "http://ftp.drupal.org/files/projects/zurb-foundation-7.x-5.0-rc6.zip" file could not be downloaded (HTTP/1.1 404 Not Found)
   Now trying to download from source
 - Installing drupal/zurb_foundation (7.5.0-rc6)
   Cloning 7bcab1aebafdcf74dcaa2a5b4a884d4ed3a7b1dc
```

**Comment by tobiasbaehr** [comment]
Take a look at https://www.drupal.org/node/2340969 then you see why it does not work.

**Comment by webflo** [comment]
We use the git repo url to generate the dist url. I think thats wrong. We have to use the project name instead.

**Comment by gapple** [comment]
The file name is based on the project name at the time it was created, so projects that had their hyphenated names fixed will have older files (created more than ~1 year ago) with the hyphen, and newer packages will use the corrected underscore.

I tried to trace down where the project name was being provided, but was not successful.
- `Drupal\ParseComposer\DrupalOrg\DistInformation` converts the project name & version into a URL
- `Drupal\ParseComposer\GitDriver::getDist()` provides the project name & version

I created drupal-composer/drupal-parse-composer#57 as a quick fix for generating the file URLs, since there could potentially be other effects from a larger change (such as the git URLs, which retain the hyphen).
